src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -564,647 +556,613 @@ namespace HVACUnitaryBypassVAV {
-                        DXCoils::GetCoilInletNode(state, CBVAV(CBVAVNum).HeatCoilType, CBVAV(CBVAVNum).HeatCoilName, DXCoilErrFlag);
-                    CBVAV(CBVAVNum).HeatingCoilOutletNode =
-                        DXCoils::GetCoilOutletNode(state, CBVAV(CBVAVNum).HeatCoilType, CBVAV(CBVAVNum).HeatCoilName, DXCoilErrFlag);
-                    if (DXCoilErrFlag) ShowContinueError(state, format("...occurs in {} \"{}\"", CBVAV(CBVAVNum).UnitType, CBVAV(CBVAVNum).Name));
+                    DXCoils::GetDXCoilIndex(state, thisCBVAV.HeatCoilName, thisCBVAV.DXHeatCoilIndexNum, DXCoilErrFlag, thisCBVAV.HeatCoilType);
+                    thisCBVAV.MinOATCompressor = DXCoils::GetMinOATCompressor(state, thisCBVAV.DXHeatCoilIndexNum, DXCoilErrFlag);
+                    thisCBVAV.HeatingCoilInletNode = DXCoils::GetCoilInletNode(state, thisCBVAV.HeatCoilType, thisCBVAV.HeatCoilName, DXCoilErrFlag);
+                    thisCBVAV.HeatingCoilOutletNode =
+                        DXCoils::GetCoilOutletNode(state, thisCBVAV.HeatCoilType, thisCBVAV.HeatCoilName, DXCoilErrFlag);
+                    if (DXCoilErrFlag) ShowContinueError(state, format("...occurs in {} \"{}\"", thisCBVAV.UnitType, thisCBVAV.Name));
 
                 } else if (UtilityRoutines::SameString(Alphas(16), "Coil:Heating:DX:VariableSpeed")) {
-                    CBVAV(CBVAVNum).HeatCoilType_Num = DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed;
+                    thisCBVAV.HeatCoilType_Num = DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed;
                     DXCoilErrFlag = false;
-                    CBVAV(CBVAVNum).DXHeatCoilIndexNum = VariableSpeedCoils::GetCoilIndexVariableSpeed(
-                        state, CBVAV(CBVAVNum).HeatCoilType, CBVAV(CBVAVNum).HeatCoilName, DXCoilErrFlag);
-                    CBVAV(CBVAVNum).MinOATCompressor =
-                        VariableSpeedCoils::GetVSCoilMinOATCompressor(state, CBVAV(CBVAVNum).DXHeatCoilIndexNum, DXCoilErrFlag);
-                    CBVAV(CBVAVNum).HeatingCoilInletNode = VariableSpeedCoils::GetCoilInletNodeVariableSpeed(
-                        state, CBVAV(CBVAVNum).HeatCoilType, CBVAV(CBVAVNum).HeatCoilName, DXCoilErrFlag);
-                    CBVAV(CBVAVNum).HeatingCoilOutletNode = VariableSpeedCoils::GetCoilOutletNodeVariableSpeed(
-                        state, CBVAV(CBVAVNum).HeatCoilType, CBVAV(CBVAVNum).HeatCoilName, DXCoilErrFlag);
-                    if (DXCoilErrFlag) ShowContinueError(state, format("...occurs in {} \"{}\"", CBVAV(CBVAVNum).UnitType, CBVAV(CBVAVNum).Name));
+                    thisCBVAV.DXHeatCoilIndexNum =
+                        VariableSpeedCoils::GetCoilIndexVariableSpeed(state, thisCBVAV.HeatCoilType, thisCBVAV.HeatCoilName, DXCoilErrFlag);
+                    thisCBVAV.MinOATCompressor = VariableSpeedCoils::GetVSCoilMinOATCompressor(state, thisCBVAV.DXHeatCoilIndexNum, DXCoilErrFlag);
+                    thisCBVAV.HeatingCoilInletNode =
```

Comment:
Same comment here. You already have the heating coil index, there is no need to find it again using name and type.